osu.Game/Rulesets/Mods/IApplicableToDrawableHitObject.cs

```diff
@@ -1,21 +1,30 @@
 ﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
 // See the LICENCE file in the repository root for full licence text.
 
+using System;
 using System.Collections.Generic;
+using osu.Framework.Extensions.IEnumerableExtensions;
 using osu.Game.Rulesets.Objects.Drawables;
 
 namespace osu.Game.Rulesets.Mods
 {
     /// <summary>
     /// An interface for <see cref="Mod"/>s that can be applied to <see cref="DrawableHitObject"/>s.
     /// </summary>
-    public interface IApplicableToDrawableHitObjects : IApplicableMod
+    public interface IApplicableToDrawableHitObject : IApplicableMod
     {
         /// <summary>
-        /// Applies this <see cref="IApplicableToDrawableHitObjects"/> to a list of <see cref="DrawableHitObject"/>s.
+        /// Applies this <see cref="IApplicableToDrawableHitObject"/> to a <see cref="DrawableHitObject"/>.
         /// This will only be invoked with top-level <see cref="DrawableHitObject"/>s. Access <see cref="DrawableHitObject.NestedHitObjects"/> if adjusting nested objects is necessary.
         /// </summary>
-        /// <param name="drawables">The list of <see cref="DrawableHitObject"/>s to apply to.</param>
+        void ApplyToDrawableHitObject(DrawableHitObject drawable);
+    }
+
+    [Obsolete(@"Use the singular version IApplicableToDrawableHitObject instead.")] // Can be removed 20211216
+    public interface IApplicableToDrawableHitObjects : IApplicableToDrawableHitObject
+    {
         void ApplyToDrawableHitObjects(IEnumerable<DrawableHitObject> drawables);
+
+        void IApplicableToDrawableHitObject.ApplyToDrawableHitObject(DrawableHitObject drawable) => ApplyToDrawableHitObjects(drawable.Yield());
```

Comment:
This is going to be a little expensive with how many single-item arrays this is going to allocate, isn't it? Not sure it's worth it for the ease of transition, given that the fix in 100% of cases is going to be implementing a different interface and removing a loop (especially that I'm not sure I necessarily agree with deprecation in the first place).

I guess something similar to this was in `Playfield` already, but if we can spare all of these allocs...

Reply:
I don't think the overhead is an issue because that is once per pooled DHO (pooled DHO), or only once before gameplay (non-pooled DHOs).

You probably remembered `Yield` in `Playfield`:
https://github.com/ppy/osu/pull/13523/files#diff-bc1ac74ef2d781278a2d0753b59b1d81b376b39bb9f2e2d3332eb151cddb312eL360

Reply:
Sure, but that's still up to 50 allocs per HO type in some cases as pool limits are intended to be absolute worst-case when it comes to the count of objects on screen. Size-wise it won't matter but count-wise it isn't that insignificant.

I'd probably add a dated comment to drop this default implementation and replace with local loops at call sites when the interface is fully removed after the 6 month period.

Reply:
I don't understand your suggestion.
The default implementation (that uses `Yield`) is in the obsolete interface, and it is not used by the game (dead code except custom rulesets).
In the code using the new `IApplicableToDrawableHitObject` interface (without suffix `s`), `Yield` is not used at all.

Reply:
Never mind, got confused. This will go away on its own with the interface removal, yes.